smarty/templates/directentry.tpl

```diff
@@ -25,7 +26,51 @@
             {$study_title}
         </th>
     </tr>
-
+        <!-- script src="{$baseurl}/js/loris.js" type="text/javascript"></script -->
+        <script src="{$baseurl}/js/modernizr/modernizr.js" type="text/javascript"></script>
+        <script type="text/javascript">
+          $(document).ready(function() {
+            {if $crumbs != "" && empty($error_message)}
+              var crumbs = {$crumbs|@json_encode},
+                      baseurl = "{$baseurl}",
+                      breadcrumbs = RBreadcrumbs({
+                        breadcrumbs: crumbs,
+                        baseURL: baseurl
+                      });
+              React.render(breadcrumbs, document.getElementById("breadcrumbs"));
+            {/if}
+            // If <input type="date" /> is not supported (i.e. Firefox), load
+            // jquery date-picker
+            if (!Modernizr.inputtypes.date) {
+              var dateInputs = $('input[type=date]');
+              dateInputs.datepicker({
+                dateFormat: 'yy-mm-dd',
+                changeMonth: true,
+                changeYear: true,
+                yearRange: "1900:" + new Date().getFullYear(),
+                constrainInput: true
+              });
+              dateInputs.attr('placeholder', 'yyyy-mm-dd');
+              dateInputs.on('keydown paste', function(e) { e.preventDefault(); });
+            }
+            if (!Modernizr.inputtypes.month) {
+              var monthInputs = $('input[type=month]');
+              monthInputs.datepicker({
+                dateFormat: 'yy-mm',
+                changeMonth: true,
+                changeYear: true,
+                yearRange: "1900:" + new Date().getFullYear(),
+                constrainInput: true,
+                onChangeMonthYear: function(y, m, d) {
+                  // Update date in the input field
+                  $(this).datepicker('setDate', new Date(y, m - 1, d.selectedDay));
+                }
+              });
+              monthInputs.attr('placeholder', 'yyyy-mm');
+              monthInputs.on('keydown paste', function(e) { e.preventDefault(); });
+            }
```

Comment:
None of this looks like it's using any smarty or anything other than plain javascript. Why is it mixed in the template file and not a self-contained `.js` file?